examples/chip-tool/commands/payload/AdditionalDataParseCommand.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "AdditionalDataParseCommand.h"
+
+#include <setup_payload/AdditionalDataPayload.h>
+#include <setup_payload/AdditionalDataPayloadParser.h>
+
+using namespace ::chip;
+using namespace ::chip::SetupPayload;
+
+CHIP_ERROR AdditionalDataParseCommand::Run(PersistentStorage & storage, NodeId localId, NodeId remoteId)
+{
+    std::vector<uint8_t> payloadData;
+    AdditionalDataPayload resultPayload;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    std::string payloadString(mPayload);
+
+    // Decode input payload
+    size_t len = payloadString.length();
+
+    for (size_t i = 0; i < len; i += 2)
+    {
+        auto str  = payloadString.substr(i, 2);
+        uint8_t x = (uint8_t) stoi(str, 0, 16);
```

Comment:
So if the string is not hex digits this will throw, right?  Probably OK, for chip-tool....

This file needs to `#include <string>`

Reply:
that's strange, it's compiling on macos without including the string header , i added it

Reply:
It's possible it included it indirectly via one of the other headers, but we shouldn't rely on that.

Reply:
added